etl/compare.py

```diff
@@ -0,0 +1,273 @@
+            preview_different_dataframe_values=show_values,
+            show_shared=show_shared,
+            truncate_lists_longer_than=truncate_lists_at,
+        )
+        for line in lines:
+            print(line)
+        if diff.are_structurally_equal:
+            exit(2)
+        else:
+            exit(3)
+
+
+@cli.command(cls=RichCommand)
+@click.argument("channel")
+@click.argument("namespace")
+@click.argument("dataset")
+@click.argument("table")
+@click.option(
+    "--absolute-tolerance",
+    default=0.00000001,
+    show_default=True,
+    help="The absolute tolerance for floating point comparisons.",
+)
+@click.option(
+    "--relative-tolerance",
+    default=0.05,
+    show_default=True,
+    help="The relative tolerance for floating point comparisons.",
+)
+@click.option(
+    "--show-values/--hide-values",
+    default=False,
+    show_default=True,
+    help="Show a preview of the values where the dataframes are different.",
+)
+@click.option(
+    "--show-shared/--hide-shared",
+    default=False,
+    show_default=True,
+    help="Show the structural overlap of the two dataframes (shared columns, index columns and index values).",
+)
+@click.option(
+    "--truncate-lists-at",
+    default=20,
+    show_default=True,
+    help="Print truncated lists if they are longer than the given length.",
+)
+def etl_catalog(
+    channel: str,
+    namespace: str,
+    dataset: str,
+    table: str,
+    absolute_tolerance: float,
+    relative_tolerance: float,
+    show_values: bool,
+    show_shared: bool,
+    truncate_lists_at: int,
+) -> None:
+    """
+    Compare a table in the local catalog with the one in the remote catalog.
+
+    It compares the columns, index columns and index values (row indices) as
+    sets between the two dataframes and outputs the differences. Finally it compares the values of the overlapping
+    columns and rows with the given threshold values for absolute and relative tolerance.
+
+    The exit code is 0 if the dataframes are equal, 1 if there is an error loading the dataframes, 2 if the dataframes
+    are structurally equal but are otherwise different, 3 if the dataframes have different structure and/or different values.
+    """
+    try:
+        remote_df = catalog.find_one(
+            table=table, namespace=namespace, dataset=dataset, channels=channel
+        )
+        local_catalog = catalog.LocalCatalog("data")
+        local_df = local_catalog.find_one(
+            table=table,
+            namespace=namespace,
+            dataset=dataset,
+            channel=cast(catalog.CHANNEL, channel),
+        )
+    except Exception as e:
+        print(e)
+        exit(1)
+
+    diff_print_and_exit(
+        remote_df,
+        local_df,
+        "remote",
+        "local",
+        absolute_tolerance,
+        relative_tolerance,
+        show_values,
+        show_shared,
+        truncate_lists_at,
+    )
+
+
+def load_table(path_str: str) -> catalog.Table:
```

Comment:
(There's a [similar method](https://github.com/owid/owid-catalog-py/blob/master/owid/catalog/catalogs.py#L329) that could be used instead, but it'd probably look weirder than your method)